src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -161,89 +203,11 @@ private static SelectionInfo AdjustFinalTokensBasedOnContext(
             return selectionInfo with
             {
                 SelectionInExpression = firstValidNode is ExpressionSyntax,
-                SelectionInSingleStatement = firstValidNode is StatementSyntax,
                 FirstTokenInFinalSpan = firstValidNode.GetFirstToken(includeZeroWidth: true),
                 LastTokenInFinalSpan = firstValidNode.GetLastToken(includeZeroWidth: true),
             };
         }
 
-        protected override SelectionInfo GetInitialSelectionInfo()
-        {
-            var root = this.SemanticDocument.Root;
-
-            var adjustedSpan = GetAdjustedSpan(OriginalSpan);
-
-            var firstTokenInSelection = root.FindTokenOnRightOfPosition(adjustedSpan.Start, includeSkipped: false);
-            var lastTokenInSelection = root.FindTokenOnLeftOfPosition(adjustedSpan.End, includeSkipped: false);
-
-            if (firstTokenInSelection.Kind() == SyntaxKind.None || lastTokenInSelection.Kind() == SyntaxKind.None)
-            {
-                return new SelectionInfo { Status = new OperationStatus(succeeded: false, FeaturesResources.Invalid_selection) };
-            }
-
-            if (firstTokenInSelection.SpanStart > lastTokenInSelection.Span.End)
-            {
-                return new()
-                {
-                    Status = new OperationStatus(succeeded: false, FeaturesResources.Selection_does_not_contain_a_valid_token),
-                    FirstTokenInOriginalSpan = firstTokenInSelection,
-                    LastTokenInOriginalSpan = lastTokenInSelection
-                };
```

Comment:
instead of having the moved code have this pattern (where it repeats the failure case multiple times), instead it just computes the failure code, and then if that happens, it creates the infla failure object once.